src/tests/Microsoft.PowerFx.Core.Tests/CheckInvocationTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Core.Tests
+{
+    public class CheckInvocationTests : PowerFxTest

```

Comment:
>CheckInvocationTests [](http://example.com/codeflow?start=17&length=20)

Add tests for:

- overloads,  (is there  case where overloads return different types?)
- namespaces. 



If() is great test because return type is dynamic.